frame/staking/src/lib.rs

```diff
@@ -2581,18 +2639,23 @@ impl<T: Config> frame_election_provider_support::ElectionDataProvider<T::Account
 			.saturating_sub(current_era_start_session_index)
 			.min(T::SessionsPerEra::get());
 
-		let session_length = T::NextNewSession::average_session_length();
-
 		let until_this_session_end = T::NextNewSession::estimate_next_new_session(now)
 			.0
 			.unwrap_or_default()
 			.saturating_sub(now);
 
-		let sessions_left: T::BlockNumber = T::SessionsPerEra::get()
-			.saturating_sub(era_length)
-			// one session is computed in this_session_end.
-			.saturating_sub(1)
-			.into();
+		let session_length = T::NextNewSession::average_session_length();
+
+		let sessions_left: T::BlockNumber = match ForceEra::get() {
+			Forcing::ForceNone => Bounded::max_value(),
+			Forcing::ForceNew | Forcing::ForceAlways => Zero::zero(),
+			Forcing::NotForcing if era_length >= T::SessionsPerEra::get() => Zero::zero(),
```

Comment:
Maybe beyond the scope of this PR but what about renaming `era_length` to `era_progress`?

Reply:
yeah I agree `era_progress` is a better name